### PR TITLE
Test Mailer NTLM authentication

### PR DESCRIPTION
**Pull request by cescoffier**
Just a test verifying the NTML authentication support of the mailer.

### CODE REVIEW

**Comment by cescoffier** [comment]
I have tested in native. It works. But I can't use my fake smtp server in a native test.

**Review by stuartwdouglas — APPROVED** [review]

**Comment by stuartwdouglas** [comment]
I think you could by just running the fake server in the test JVM and having the native image connect to it.

File: extensions/mailer/deployment/src/test/java/io/quarkus/mailer/FakeSmtpServer.java [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    }
+
+    public FakeSmtpServer setCloseWaitTime(int time) {
+        closeWaitTime = time;
+        return this;
+    }
+
+    public void stop() {
+        if (netServer != null) {
+            CountDownLatch latch = new CountDownLatch(1);
```
Comment (stuartwdouglas) [inline extensions/mailer/deployment/src/test/java/io/quarkus/mailer/FakeSmtpServer.java:177]: This latch is not used

File: extensions/mailer/deployment/src/test/java/io/quarkus/mailer/FakeSmtpServer.java [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+import io.vertx.core.net.NetServerOptions;
+import io.vertx.core.parsetools.RecordParser;
+import io.vertx.mutiny.core.Vertx;
+import io.vertx.mutiny.core.net.NetServer;
+import io.vertx.mutiny.core.net.NetSocket;
+
+public class FakeSmtpServer {
+
+    private NetServer netServer;
+    private String[][] dialogue;
```
Comment (stuartwdouglas) [inline extensions/mailer/deployment/src/test/java/io/quarkus/mailer/FakeSmtpServer.java:18]: I think these should be volatile, as they are set from the test thread but read from the IO thread.

**Review by stuartwdouglas — CHANGES_REQUESTED** [review]
Actually I just noticed a thread safety issue

**Pull request closed without merging**